openapi-ls-extension/src/main/java/io/ballerina/openapi/extension/OpenAPIConverterService.java

```diff
@@ -152,16 +152,17 @@ public CompletableFuture<OpenAPIConverterResponse> generateOpenAPI(OpenAPIConver
                 return response;
             }
             Module defaultModule = module.get().currentPackage().getDefaultModule();
-
+            SemanticModel updatedSemanticModel = null;
             JsonArray specs = new JsonArray();
             for (DocumentId currentDocumentID : defaultModule.documentIds()) {
                 Document document = defaultModule.document(currentDocumentID);
                 Optional<Path> path = defaultModule.project().documentPath(currentDocumentID);
                 Path inputPath = path.orElse(null);
                 SyntaxTree syntaxTree = document.syntaxTree();
+                updatedSemanticModel = updatedSemanticModel != null ? updatedSemanticModel :
+                        module.get().currentPackage().getCompilation().getSemanticModel(currentDocumentID.moduleId());
```

Comment:
Shall we get the semantic module by directly using the default module ID instead?

Reply:
Added